src/components/SingleArticle.jsx

```diff
@@ -0,0 +1,23 @@
+import { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { fetchArticleById } from "./Api";
+
+const SingleArticle = () => {
+  const [article, setArticle] = useState({});
+  const { article_id } = useParams();
+
+  useEffect(() => {
+    fetchArticleById(article_id).then((data) => {
+      setArticle(data.article);
+    });
+  }, []);
```

Comment:
Could we add some loading state here, just incase the api takes a while to respond?